### DISCUSSION TITLE
[categorie_corrigee] "coherence": petite variation dans l'algorithme du Cas 3 par rapport au Cas 1 et Cas 2

### DISCUSSION THREAD

**Discussion by JMPivette** [JMPivette/evavelo#7] [Q&A] [answered]
@VeloTerritoires

Dans le chapitre 3.1.11, les Cas 1, 2 et 3 sont a peu près identiques mis à part une petite subtilité:

Cas 1 et 2:
[...]Si le type de sortie déclarée n’est pas « une randonnée sur plusieurs jours » et qu’il y a des réponses aux questions spécifiques des itinérants, alors le cycliste sera attribué à la catégorie [...]

Cas 3:
[...]Si le type de sortie déclarée n’est pas « une randonnée sur plusieurs jours » et la durée de séjour supérieure à 1 jour et/ou que des réponses **cohérentes** ont été apportées aux questions spécifiques des itinérants, alors le cycliste sera attribué à la catégorie « itinérant »[...]

Est-ce qu'il y a un critère pour des réponses _cohérentes_ ou est-ce qu'on considère que la moindre réponses aux questions spécifiques itinérants est OK?

**Comment by SteffMangin** [comment]
En effet, bonne question. La cohérence pourrait être :
- la présence de réponse à au moins deux des trois questions : 
       * [iti_km_voyage], 
       * [iti_depart_initial], [iti_depart_itineraire], iti_arrivee_itineraire], [iti_arrivee_final],
       * [iti_experience]
- une distance parcourue pendant l'ensemble du voyage [iti_km_voyage] / [dms] > 40 km
- des communes de départ et d'arrivée sur l'itinéraire qui appartiennent à l'itinéraire

Dis-moi si c'est trop compliqué ?

**Reply by JMPivette** [reply]
Est-ce que cette notion de **"cohérence"** est spécifique aux cas **3 et 10** ou s'appliquent également à l'ensemble des algorithmes similaires (1; 2; 3; 4; 7; 10) ?

Faut-il que les 3 critères soient respectés ou seulement l'un de ces critères?

**Comment by SteffMangin** [comment]
Bonjour Jean Marie, 

J'ai un souci avec mon pc et je ne peux pas le régler ce matin. Je te réponds donc par mail... Et de mémoire. 
Si les cas que ti cites sont tous ceux où il y a consultation des infos itinerance, alors oui il serait plus cohérent :-) d'appliquer la cohérence partout. 
Je ne me souviens plus des critères que je t'avais donnés(vive les vacances)... Je ccrois que c'était la distance et des villes compatibles avec une itinerance ??? Je dirai qu'une reponse suffit. Mais cela me fait penser qu'il ne doit pas y avoir de contrôles de cette coherence dans les controles embarqués pour vérifier la qualité des reponses.

**Comment by JMPivette** [comment]
Pour résumer:
La "cohérence" des réponses spécifiques itinérant est à appliquer à l'ensemble des cas similaires: (1; 2; 3; 4; 7; 10).

Cohérence:
* Une distance parcourue pendant l'ensemble du voyage [iti_km_voyage] / [dms] > 40 km

OU
* Des communes de départ et d'arrivée sur l'itinéraire qui appartiennent à l'itinéraire 
         ET 
   La présence de réponse à au moins deux des trois questions :
    * [iti_km_voyage],
    * [iti_depart_initial], [iti_depart_itineraire], [iti_arrivee_itineraire], [iti_arrivee_final],
    * [iti_experience]

**Reply by SteffMangin** [reply]
yes ;-)

**Comment by JMPivette** [comment, marked as answer]
Critère de cohérence modifié suite à discussion sur #39 :

Cohérence:
* Une distance parcourue pendant l'ensemble du voyage [iti_km_voyage] / [dms] > 40 km

OU
* La présence de réponses à au moins deux des trois critères suivants:
  * [iti_km_voyage]
  * [iti_experience]
  * Critère départ/arrivée:
    * [iti_depart_initial] OU [iti_dep_iti_valide]
    ET
    * [iti_arrivee_itineraire] OU [iti_arr_iti_valide]
